pennylane/operation.py

```diff
@@ -364,6 +365,20 @@ def __init__(self, *params, wires=None, do_queue=True):
         if wires is None:
             raise ValueError("Must specify the wires that {} acts on".format(self.name))
 
+        # turn wires into Wires object
+        self._wires = Wires(wires)  #: Wires: wires on which the operator acts
+
+        # check that the number of wires given corresponds to required number
+        if (
+            self.num_wires != AllWires
+            and self.num_wires != AnyWires
+            and len(self._wires) != self.num_wires
+        ):
+            raise ValueError(
+                "{}: wrong number of wires. "
+                "{} wires given, {} expected.".format(self.name, len(wires), self.num_wires)
```

Comment:
I think we want to avoid using `len(wires)` here, since a string would give a confusing error message:
```
qml.RX(0.5, wires="qubit0")  # len(wires)==6
```
```suggestion
                "{} wires given, {} expected.".format(self.name, len(self._wires), self.num_wires)
```